### PR TITLE
address dns metrics comments

### PR DESCRIPTION
**Pull request by ramaraochavali**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn PTAL

File: pilot/pkg/dns/monitoring.go [hunk 1]
```diff
@@ -30,14 +30,14 @@ var (
...
-		"dns_total_upstream_failures",
+		"dns_upstream_failures_total",
 		"Total number of DNS requests forwarded to upstream.",
 	)
 
 	requestDuration = monitoring.NewDistribution(
-		"dns_upstream_request_time_ms",
-		"Total time in milliseconds Istio takes to get DNS response from upstream.",
-		[]float64{1, 3, 5, 10, 25, 50},
+		"dns_upstream_request_time",
```
Comment (howardjohn) [inline pilot/pkg/dns/monitoring.go:38]:
```suggestion
		"dns_upstream_request_duration_seconds",
```

File: pilot/pkg/dns/monitoring.go [hunk 2]
```diff
@@ -30,14 +30,14 @@ var (
...
+		"Total time in seconds Istio takes to get DNS response from upstream.",
+		[]float64{.01, .1, 0.5, 1, 3, 5},
```
Comment (howardjohn) [inline pilot/pkg/dns/monitoring.go:40]: The buckets seem pretty skewed to extreme latency, while before we had max 50ms. I think we could probably have something like:
`[]float64{.005, .001, 0.01, 0.1, 1, 5},`

WDYT?

File: pilot/pkg/dns/monitoring.go [hunk 3]
```diff
@@ -30,14 +30,14 @@ var (
 	)
 
 	failures = monitoring.NewSum(
-		"dns_total_upstream_failures",
+		"dns_upstream_failures_total",
 		"Total number of DNS requests forwarded to upstream.",
 	)
 
 	requestDuration = monitoring.NewDistribution(
```
Comment (howardjohn) [inline pilot/pkg/dns/monitoring.go:37]: We need to change the .Record call to use seconds now I think

File: pilot/pkg/dns/monitoring.go (hunk 2 above)
Comment (ramaraochavali) [reply pilot/pkg/dns/monitoring.go:40]: Makes sense will change

File: pilot/pkg/dns/monitoring.go (hunk 3 above)
Comment (ramaraochavali) [reply pilot/pkg/dns/monitoring.go:37]: correct. Missed that. Thanks for catching it.

**Comment by ramaraochavali** [comment]
@howardjohn fixed.

**Review by howardjohn — APPROVED** [review]
for future viewers - this is backwards compatible because we never shipped the original names

**Pull request merged**